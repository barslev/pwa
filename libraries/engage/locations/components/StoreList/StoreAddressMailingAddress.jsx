import React from 'react';
import PropTypes from 'prop-types';
import { LocationIcon } from '../../../components';
import { i18n } from '../../../core';
import { address as container, addressIcon } from './style';

/**
 * Renders the a pickup location's address information.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
function StoreAddressMailingAddress({ address }) {
  if (!address) {
    return null;
  }

  return (
    <div className={container}>
      <div className={addressIcon}>
        <LocationIcon />
      </div>
      <div>
        <div data-test-id="street">
          {address.street}
        </div>
        {(address.street2 && address.street2 !== '') && (
          <div data-test-id="street2">
            {address.street2}
          </div>
        )}
        {(address.street3 && address.street3 !== '') && (
          <div data-test-id="street3">
            {address.street3}
          </div>
        )}
        {(address.street4 && address.street4 !== '') && (
          <div data-test-id="street4">
            {address.street4}
          </div>
        )}
        {i18n.text('product.location.address', address)}
      </div>
    </div>
  );
}

StoreAddressMailingAddress.propTypes = {
  address: PropTypes.shape({
    city: PropTypes.string,
    postalCode: PropTypes.string,
    region: PropTypes.string,
    street: PropTypes.string,
    street2: PropTypes.string,
    street3: PropTypes.string,
    street4: PropTypes.string,
  }),
};

StoreAddressMailingAddress.defaultProps = {
  address: null,
};

export default StoreAddressMailingAddress;
