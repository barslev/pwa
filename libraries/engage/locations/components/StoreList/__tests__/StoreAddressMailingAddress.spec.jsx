import React from 'react';
import { shallow } from 'enzyme';
import StoreAddressMailingAddress from '../StoreAddressMailingAddress';

const dummyAddress = {
  street: 'Test Street 123',
  city: 'Testcity',
  postalCode: '12345',
  region: 'TX',
  street3: '',
};

describe('<StoreAddressMailingAddress />', () => {
  it('should not render if no address is passed in', () => {
    const wrapper = shallow(<StoreAddressMailingAddress />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance()).toEqual(null);
  });

  it('should render as expected', () => {
    const wrapper = shallow(<StoreAddressMailingAddress address={dummyAddress} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('LocationIcon')).toBeTruthy();
    expect(wrapper.find('[data-test-id="street"]').text()).toEqual(dummyAddress.street);
    expect(wrapper.find('[data-test-id="street2"]').getElements().length).toEqual(0);
    expect(wrapper.find('[data-test-id="street3"]').getElements().length).toEqual(0);
    expect(wrapper.find('[data-test-id="street4"]').getElements().length).toEqual(0);
    expect(wrapper.find('Translate')).toBeTruthy();
  });

  it('should render more streets if necessary', () => {
    const wrapper = shallow(<StoreAddressMailingAddress
      address={{
        ...dummyAddress,
        street2: 'Some additional street info',
      }}
    />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('LocationIcon')).toBeTruthy();
    expect(wrapper.find('[data-test-id="street"]').text()).toEqual(dummyAddress.street);
    expect(wrapper.find('[data-test-id="street2"]').text()).toEqual('Some additional street info');
    expect(wrapper.find('[data-test-id="street3"]').getElements().length).toEqual(0);
    expect(wrapper.find('[data-test-id="street4"]').getElements().length).toEqual(0);
    expect(wrapper.find('Translate')).toBeTruthy();
  });
});
