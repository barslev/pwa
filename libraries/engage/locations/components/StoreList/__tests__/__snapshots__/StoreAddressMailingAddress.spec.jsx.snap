// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoreAddressMailingAddress /> should not render if no address is passed in 1`] = `""`;

exports[`<StoreAddressMailingAddress /> should render as expected 1`] = `
<div
  className={
    Object {
      "data-css-1npkmgj": "",
    }
  }
>
  <div
    className={
      Object {
        "data-css-1rqdg8e": "",
      }
    }
  >
    <Location />
  </div>
  <div>
    <div
      data-test-id="street"
    >
      Test Street 123
    </div>
    locations.address
  </div>
</div>
`;

exports[`<StoreAddressMailingAddress /> should render more streets if necessary 1`] = `
<div
  className={
    Object {
      "data-css-1npkmgj": "",
    }
  }
>
  <div
    className={
      Object {
        "data-css-1rqdg8e": "",
      }
    }
  >
    <Location />
  </div>
  <div>
    <div
      data-test-id="street"
    >
      Test Street 123
    </div>
    <div
      data-test-id="street2"
    >
      Some additional street info
    </div>
    locations.address
  </div>
</div>
`;
